src/server/summary-log-upload/controller.test.js

```diff
@@ -71,6 +71,42 @@ describe('#summaryLogUploadController', () => {
 
     expect(result).toEqual(expect.stringContaining('Summary log upload error'))
   })
+
+  test('should display formErrors when lastError exists in session', async () => {
+    const errorMessage = 'Something went wrong'
+    yar.get.mockImplementation(() => ({ lastError: errorMessage }))
+    overrideRequest(server, yar)
+
+    const { result, statusCode } = await server.inject({ method: 'GET', url })
+
+    expect(result).toEqual(expect.stringContaining(errorMessage))
+    expect(statusCode).toBe(statusCodes.ok)
+  })
+
+  test('should clear lastError after it has been read once', async () => {
+    const errorMessage = 'Flash error'
+    const sessionData = { lastError: errorMessage }
+    yar.get.mockImplementation(() => sessionData)
+    overrideRequest(server, yar)
+
+    await server.inject({ method: 'GET', url })
+
+    expect(yar.set).toHaveBeenCalledWith(
+      'summaryLogs',
+      expect.not.objectContaining({ lastError: errorMessage })
+    )
+  })
+
+  test('should not set lastError if none exists', async () => {
+    yar.get.mockImplementation(() => ({}))
+    overrideRequest(server, yar)
+
+    await server.inject({ method: 'GET', url })
+    const callArgs = yar.set.mock.calls.find(
+      ([key]) => key === 'summaryLogs'
+    )[1]
```

Comment:
I think you can use `.toBeNthCalledWith(n, 'summaryLogs', expect.objectContaining({ lastError: undefined }))` to solve this assertion